src/psyclone/tests/psyir/transformations/increase_rank_loop_arrays_test.py

```diff
@@ -0,0 +1,130 @@
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: S. Siso, STFC Daresbury Lab
+
+''' Module containing tests for the IncreaseRankLoopArraysTrans class. '''
+
+import pytest
+from psyclone.psyir.nodes import Loop
+from psyclone.psyir.transformations import IncreaseRankLoopArraysTrans
+from psyclone.transformations import TransformationError
+
+
+def test_trans_name_and_str():
+    ''' Check that the name and str method works as expected. '''
+    trans = IncreaseRankLoopArraysTrans()
+    assert trans.name == "IncreaseRankLoopArraysTrans"
+    assert str(trans) == (
+        "Increases the Rank of the supplied arrays by the iteration space of "
+        "the given loop, and update all its references")
+
+
+def test_irla_validate(fortran_reader):
+    ''' Check that the validate method works as expected. '''
+    trans = IncreaseRankLoopArraysTrans()
+
+    psyir = fortran_reader.psyir_from_source("""
+        program test
+            real, dimension(10) :: a
+            integer :: i = 3
+            a(i) = 3
+            do i = 1, 10
+               a(i) = 3
+            end do
+            associate(x => a(i))
+            end associate
+        end program
+    """)
+
+    routine = psyir.children[0]
+    with pytest.raises(TransformationError) as err:
+        trans.apply(routine.children[0])
+    assert ("The target of the IncreaseRankLoopArraysTrans transformation "
+            "should be a Loop, but found 'Assignment'." in str(err.value))
+
+    with pytest.raises(TransformationError) as err:
+        trans.apply(routine.children[1])
+    assert ("IncreaseRankLoopArraysTrans has a mandatory 'arrays' option "
+            "that needs to be provided to inform what arrays needs their "
+            "rank increased." in str(err.value))
+
+    with pytest.raises(TransformationError) as err:
+        trans.apply(routine.children[1], options={'arrays': ['i']})
+    assert ("IncreaseRankLoopArraysTrans provided 'arrays' must be a local "
+            "array symbols, but 'i: " in str(err.value))
+
+    with pytest.raises(TransformationError) as err:
+        trans.apply(routine.children[1], options={'arrays': ['a']})
+    assert ("IncreaseRankLoopArraysTrans does not support arrays that are "
+            "referenced inside a Codeblock, but 'a' is inside one."
+            in str(err.value))
+
+    # Referencing non-existan arrays is fine, they are just ignored, this
```

Comment:
Please add this information to the docstring of the transformation.

Reply:
This is not the case anymore